### PR TITLE
add r-lyeh-archived/sole

### PR DESCRIPTION
**Pull request by xyz1001**
Specify library name and version:  **sole/1.0.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

This library is a dependency of my project.
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sole/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        os.rename(extracted_name, self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern="sole.hpp", dst="include",
+                  src=self._source_subfolder)
+
+    def package_info(self):
+        if str(self.settings.os) in ["Linux"]:
```
Comment (theirix) [inline recipes/sole/all/conanfile.py:35]: it's enough to have:

```suggestion
        if self.settings.os == "Linux":
```

**Review by theirix — DISMISSED** [review]

File: recipes/sole/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_name = "sole-" + self.version
+        os.rename(extracted_name, self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern="sole.hpp", dst="include",
+                  src=self._source_subfolder)
```
Comment (SpaceIm) [inline recipes/sole/all/conanfile.py:32]: `sole.fwd.hpp` should also be packaged.

Comment (xyz1001) [reply recipes/sole/all/conanfile.py:32]: fixed

**Review by SpaceIm — APPROVED** [review]

**Review by theirix — APPROVED** [review]

File: recipes/sole/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+class SoleConan(ConanFile):
+    name = "sole"
+    homepage = "https://github.com/r-lyeh-archived/sole"
+    description = "Sole is a lightweight C++11 library to generate universally unique identificators (UUID), both v1 and v4."
+    topics = ("conan", "uuid", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler"
+    no_copy_source = True
+    license = "zlib"
```
Comment (uilianries) [inline recipes/sole/all/conanfile.py:13]:
```suggestion
    license = "Zlib"
```

https://spdx.org/licenses/Zlib

**Review by uilianries — COMMENTED** [review]
I didn't know about Sole, it looks nice!

File: recipes/sole/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern="*.hpp", dst="include",
+                  src=self._source_subfolder)
+
+    def package_info(self):
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append('rt')
```
Comment (madebr) [inline recipes/sole/all/conanfile.py:36]:
```suggestion
            self.cpp_info.system_libs.append("rt")
```

File: recipes/sole/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        os.rename(extracted_name, self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        self.copy(pattern="*.hpp", dst="include",
+                  src=self._source_subfolder)
+
+    def package_info(self):
+        if self.settings.os == "Linux":
```
Comment (madebr) [inline recipes/sole/all/conanfile.py:35]:
```suggestion
        if self.settings.os in ("FreeBSD", "Linux"):
```

**Review by madebr — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Pull request merged**